src/app/modules/hospital-map/services/building-map.service.ts

```diff
@@ -0,0 +1,62 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import * as d3 from 'd3';
+import { Observable } from 'rxjs';
+import { IBuilding } from '../model/building.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class BuildingMapService {
+
+  apiHost: string = 'http://localhost:5000/';
+  headers: HttpHeaders = new HttpHeaders({ 'Content-Type': 'application/json' });
+
+  constructor(private http: HttpClient) { }
+
+  getBuildings(): Observable<IBuilding[]> {
+    return this.http.get<IBuilding[]>(this.apiHost + 'api/intranet/map/buildings', {headers: this.headers});
+  }
+  getBuilding(id: number): Observable<IBuilding> {
+    return this.http.get<IBuilding>(this.apiHost + 'api/intranet/buildings/' + id, {headers: this.headers});
+  }
+  updateBuilding(building: any): Observable<any> {
+    return this.http.put<any>(this.apiHost + 'api/intranet/buildings/' + building.id, building.name, {headers: this.headers});
+  }
+  createSVG(){
+    return d3.select("#svgDiv").append("svg").attr("height", 500).attr("width", 800)
+  }
+  createRectangles(svg:any, data2:any){
+    return svg.selectAll("rect").data(data2).enter().append("rect")
+    .attr("height", function(d:any){ return d.height;})
+    .attr("width", function(d:any){ return d.width;})
+    .attr("fill", 'white')
+    .attr("stroke", "black")
+    .attr("x", function(d:any){ return d.xCoordinate })
```

Comment:
Instead of using function(parametar) { return ...}, you can use (parameter) => ...